src/vm/ast/definition_sorter/mod.rs

```diff
@@ -213,12 +213,11 @@ impl<'a> DefinitionSorter {
                             }
                         } else if let Some(native_function) =
                             // NOTE: can use ClarityVersion::latest() here only as long as NO NEW FUNCTIONS are special cased
-                            //        in the definition sorter.
+                        //        in the definition sorter.
                             NativeFunctions::lookup_by_name_at_version(
-                                    function_name,
-                                    &ClarityVersion::latest(),
-                                )
-                        {
+                            function_name,
+                            &ClarityVersion::latest(),
```

Comment:
Hold up -- this needs to be gated. You should be using whatever `ClarityVersion` the outermost runtime context is using.

Reply:
Hmm.. this change was not supposed to be part of this PR. It snuck in as because of some weird behavior of "cargo fmt". @kantai ?

Reply:
Right -- so the `definition_sorter` is a bit of a special case when it comes to the analysis -- it doesn't actually do any special checks for native functions vs. defined functions except for a few special cases. As long as no new special cases are added in Clarity2, it's fine to just use `::latest` here. It may be safer in the long run to gate this, but I don't feel strongly that it needs to be done in this PR.

Reply:
It probably shouldn't be in this PR, but should we make an issue for it and make sure to do it?

Reply:
Issue created #2759